doc/developer/faq.rst

```diff
@@ -0,0 +1,110 @@
+.. _contributing_faq:
+
+Contributing FAQ
+****************
+
+A collection of frequently-asked questions related to open-source development
+and contributing to NetworkX.
+
+Q: I'm new to open source and would like to contribute to NetworkX. How do I get started?
+-----------------------------------------------------------------------------------------
+
+To contribute to NetworkX, you will need three things:
+
+  1. The source code
+  2. A development environment
+  3. An idea of what you'd like to contribute
+
+Steps 1 & 2 are covered extensively in :ref:`Development Workflow <dev_workflow>`.
+There is no generic answer for step 3. There are many ways that NetworkX can
+be improved, from adding new algorithms, improving existing algorithms,
+improving the test suite (e.g. increasing test coverage), and improving the
+documentation.
+The "best" way to find a place to start is to follow your own personal
+interests!
+
+Q: How do I contribute an example to the Gallery?
+-------------------------------------------------
+
+The example gallery is great place to contribute, particularly if you have an
+interesting application or visualization that uses NetworkX.
+The gallery is generated using :doc:`sphinx-gallery <sphinx-gallery:index>`
+from Python scripts stored in the ``examples/`` directory.
+
+Example
+~~~~~~~
+
+Let's say I'd like to contribute an example of visualizing a
+`complete graph <networkx.generators.classic.complete_graph>` using a
+`circular layout <networkx.drawing.layout.circular_layout>`.
+Assuming you have already followed the procedure for
+:ref:`setting up a development environment <dev_workflow>`, start by
+creating a new branch:
+
+.. code-block:: bash
+
+   git checkout -b complete-graph-circular-layout-example
+
+.. note:: It's generally a good idea to give your branch a descriptive name so
+   that it's easy to remember what you are working on.
+
+Now you can begin work on your example. Sticking with the circular layout idea,
+you might create a file in ``examples/drawing`` called ``circular_layout.py``
```

Comment:
Should probably be called ``plot_circular_layout.py``.

Reply:
Good catch!